### PR TITLE
fix: useBackendAIImageMetaData test code related to `waitFor`

### PR DESCRIPTION
**Pull request by yomybaby**
### TL;DR

Refactored useBackendAIImageMetaData hook and improved its test suite.

This will solve [the GitHub Action failure.](https://github.com/lablup/backend.ai-webui/runs/29470763524)

### What changed?

- Simplified the useSuspenseTanQuery in useBackendAIImageMetaData hook by removing unnecessary type casting and promise chaining.
- Updated the test suite for useBackendAIImageMetaData:
  - The handler of the `waitFor` function in `@testing-library/react` should return a promise if it's an async action. [ref](https://testing-library.com/docs/dom-testing-library/api-async/#waitfor)
  - For the `waitFor`, use `toBeTruthy` instead of `toBeDefined` because `expect(null).toBeDefined();` pass the test. 

### How to test?

1. Run the existing test suite for useBackendAIImageMetaData.
2. Verify that all tests pass and cover the expected functionality.
3. Manually test the useBackendAIImageMetaData hook in the application to ensure it still functions correctly with the refactored code.

### CODE REVIEW

**Comment by yomybaby** [comment]
* **#2674** <a href="https://app.graphite.dev/github/pr/lablup/backend.ai-webui/2674?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @yomybaby and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Review by yomybaby — COMMENTED** [review]
This PR is to fix the root cause that intermittently fails CI tests. Since this is a hotfix, it will be merged without a review.

**Review by ironAiken2 — APPROVED** [review]
LGTM
